tango/integrations/torch/train.py

```diff
@@ -535,13 +543,14 @@ def save_state(step: int):
     for callback in callbacks:
         callback.pre_train_loop()
 
-    train_batch_iterator = Tqdm.tqdm(
+    train_batch_iterator_tqdm = Tqdm.tqdm(
         training_batches,
         desc="Training",
         initial=start_step,
         total=config.train_steps,
         disable=not config.is_local_main_process,
     )
+    train_batch_iterator = more_itertools.peekable(train_batch_iterator_tqdm)
```

Comment:
What's with this?

Reply:
I need to be able to look one batch ahead to find out whether this is the last batch of the epoch.

Reply:
Oh, nevermind, I see